### PR TITLE
Update distributed/CONTRIBUTING.md to remove ProcessGroupAgent references and add test instructions

### PR DESCRIPTION
**Pull request by H-Huang**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #78658 Reenable assert after test update
* **#78625 Update distributed/CONTRIBUTING.md to remove ProcessGroupAgent references and add test instructions**

ProcessGroupAgent was deprecated so these references to the tests should be removed from comments and docs.

cc @yaroslavvb

### CODE REVIEW

**Review by mrshenli — APPROVED** [review]
Thanks for updating

File: torch/csrc/autograd/python_variable.cpp [hunk 1]
```diff
@@ -462,37 +462,6 @@ static int THPVariable_clear(THPVariable* self) {
     //        because Tensor asked us to (it's already destructing).
 
     if (!self->cdata.unsafeIsBorrowed()) {
-      // TODO: empirically, on OS X this assert appears to be untrue
```
Comment (albanD) [inline torch/csrc/autograd/python_variable.cpp:465]: Why did you remove this comment?

**Review by albanD — CHANGES_REQUESTED** [review]

File: torch/csrc/autograd/python_variable.cpp (hunk 1 above)
Comment (H-Huang) [reply torch/csrc/autograd/python_variable.cpp]: I removed this for this PR and reenabled the assert in a separate PR (#78658)

**Review by albanD — APPROVED** [review]
Updated version sounds good!
